scripts/inline_comments_data_collection.py

```diff
@@ -0,0 +1,172 @@
+        save_comments_to_file(patch_id, comments)
+    return
+
+
+def save_comments_to_file(patch_id, comments):
+    resolved_comments = [comment for comment in comments if comment.is_done]
+
+    file_path = f"comments/{patch_id}.json"
+    if os.path.exists(file_path) or not resolved_comments:
+        return
+
+    with open(file_path, "w") as f:
+        json.dump([comment.__dict__ for comment in resolved_comments], f, indent=4)
+
+
+def find_recent_update(transactions, comment_date_modified):
+    updates = [
+        transaction
+        for transaction in transactions
+        if transaction["type"] == "update"
+        and transaction["dateModified"] <= comment_date_modified
+    ]
+    if not updates:
+        return None
+    most_recent_update = max(
+        updates, key=lambda transaction: transaction["dateModified"]
+    )
+    return most_recent_update
+
+
+def save_to_dataset(data):
+    dataset_file = "dataset/inline_comment_dataset.json"
+    with open(dataset_file, "a") as f:
+        f.write(json.dumps(data) + "\n")
+
+
+def to_int(value):
+    if not value:
+        return None
+    if not isinstance(value, int):
+        return int(value)
+    return value
+
+
+def process_comments(patch_threshold, diff_length_threshold):
+    comments_dir = "comments"
+    patch_count = 0
+    for file_name in os.listdir(comments_dir):
+        file_path = os.path.join(comments_dir, file_name)
+        with open(file_path, "r") as f:
+            patch_id = int(file_name.replace(".json", ""))
+            transactions = find_transactions_from_patch(patch_id)
+
+            comments = json.load(f)
+            for comment in comments:
+                comment_date_modified = comment["date_modified"]
+                most_recent_update = find_recent_update(
+                    transactions, comment_date_modified
+                )
+                if not most_recent_update:
+                    continue
+
+                fix_patch_id, revision_phid = get_diff_info_from_phid(
+                    most_recent_update["fields"]["new"]
+                )
+
+                # If the most recent patch is the original patch itself, skip it
+                if fix_patch_id == patch_id:
+                    continue
+
+                bug_id = find_bugid_from_revision_phid(phid=revision_phid)
+                review_data.load_patch_by_id(fix_patch_id)
+
+                with open(f"patches/{fix_patch_id}.patch", "r") as f:
+                    patch_diff = f.read()
+
+                if len(patch_diff) > diff_length_threshold:
+                    continue
+
+                data = {
+                    "bug_id": to_int(bug_id),
+                    "revision_phid": revision_phid,
+                    "initial_patch_id": to_int(patch_id),
+                    "fix_patch_id": to_int(fix_patch_id),
+                    "comment": comment,
+                    "fix_patch_diff": patch_diff,
+                }
+                save_to_dataset(data)
```

Comment:
This way, we open and close the file for each data point we want to save, which adds unnecessary I/O overhead.

Reply:
I've changed it so we leave the dataset file open while the script is being run: 12a1e6d1f39b0ede21d2805343f8bc54899c135c